components/suggest/src/schema.rs

```diff
@@ -21,6 +21,15 @@ pub const SQL: &str = "
         PRIMARY KEY (keyword, suggestion_id)
     ) WITHOUT ROWID;
 
+    CREATE TABLE pocket_keywords( 
+        keyword_prefix TEXT NOT NULL, 
+        keyword_suffix TEXT NOT NULL DEFAULT '', 
+        confidence INTEGER NOT NULL,
+        rank INTEGER NOT NULL,
+        suggestion_id INTEGER NOT NULL REFERENCES suggestions(id), 
+        PRIMARY KEY (keyword_prefix, keyword_suffix, suggestion_id) 
```

Comment:
Oh no, GitHub ate my comment! 😭 Posting it again...

I tested with our live Remote Settings dataset, and it failed to ingest a suggestion that had the same keyword (`"friendship expert"`) in `low_confidence_keywords` and `high_confidence_keywords`:

```
DownloadedPocketSuggestion {
    url: "https://getpocket.com/collections/pocket-best-of-2022-marisa-franco-friendship",
    title: "Best of 2022: Dr. Marisa G Franco on the Importance of Friendship",
    low_confidence_keywords: ["friendship", "friendship expert", "marisa franco", "marisa g franco", "social connections"],
    high_confidence_keywords: ["are friends important", "friendship benefits", "friendship expert", "friendship support", "how important are friends", "importance of friendship", "quality friendships", "the importance of friendship", "the power of friendship", "why you need friends"],
    score: 0.25
}
```

Per [this comment from @0c0w3](https://github.com/mozilla/application-services/pull/5841#issuecomment-1736336089), it sounds like we do want to support these. Is that right, Drew?

If so, let's make `confidence` part of the primary key:

```suggestion
        PRIMARY KEY (keyword_prefix, keyword_suffix, confidence, suggestion_id)
```

Could you also please fix up the trailing whitespace on lines 25, 26, and 29, @tiftran? Thank you! 🙏🏼 (If you haven't already set it, VS Code's `files.trimTrailingWhitespace` setting is really helpful!)

Reply:
> Per [this comment from @0c0w3](https://github.com/mozilla/application-services/pull/5841#issuecomment-1736336089), it sounds like we do want to support these. Is that right, Drew?

It's not a requirement, and re-reading my previous comment I'm not sure I agree with it on second thought, sorry! Having been involved in the process of picking these keywords, I'm pretty sure this kind of duplication is not intentional. Unless it's trivial to support in the Rust implementation, I think we should probably remove these kinds of duplicates at ingestion time too.

We could check with Nive for each of these individual cases, but right now desktop will show them as high-confidence top pick matches since it always looks up high-confidence keywords first, and I think it would be OK for us to assume that high confidence is intended.

Reply:
I've uploaded new suggestions to stage and prod with these dupes removed from `lowConfidenceKeywords` (awaiting review) and made https://github.com/mozilla-services/merino-py/pull/434 to fix the uploader.

Reply:
Duplicates are removed at ingestion time so the current schema should be fine now. Going to mark this as resolved.